rfcs/Moving o3de objects preserving history.md

```diff
@@ -0,0 +1,194 @@
+git remote add OpenXRTest ../gallifrey-OpenXRTest
+git remote add OpenXRVk ../gallifrey-OpenXRVk
+git remote update
+```
+Create a new branch to merge into allowing unrelated histories.
+```
+git switch -C xr_merge_branch
+git merge --allow-unrelated-histories XR/main
+git merge --allow-unrelated-histories OpenXRTest/main
+git merge --allow-unrelated-histories OpenXRVk/main
+```
+Then push the branch:
+```
+git push origin xr_merge_branch
+```
+__Note: If there are LFS objects we need to upload them__
+
+We are done with these remotes, so we can remove them from o3de-extras-merged.
+```
+git remote remove gallifrey-XR
+git remote remove gallifrey-OpenXRTest
+git remote remove gallifrey-OpenXRVk
+```
+Then we create our PR and we're done.
+
+## Moving a gem from the o3de/o3de to the o3de/o3de-extras
+Now that we understand what is going on lets move a gem from the core o3de/o3de repo to the extras o3de/extras repo. We will do the same thing as above, but for just one object this time.
+
+For this example we will move (https://github.com/o3de/o3de.git)\Gems\Achievements to (https://github.com/o3de/o3de-extras.git)\Gems\Achievements.
+
+[O3DE excepts pull requests from forks](https://docs.o3de.org/docs/contributing/to-code/git-workflow/). For the example I'll use my forks:
+
+* https://github.com/byrcolin/o3de.git
+
+* https://github.com/byrcolin/o3de-extras.git
+
+We need a ___fresh clone___ of our ___forks___ and since we'll only be doing just one object we wont copy it, we will just clone it into a folder named achievements-gem and achievements-extras to keep it straight.
+```
+git clone achievements-gem https://github.com/byrcolin/o3de.git
+git clone achievements-extras https://github.com/byrcolin/o3de-extras.git
+```
+In this case the relative source path "Gems/Achievements" and the destination relative path "Gems/Achievements" are the same, so we do not need to ___move___ the folder when we filter the repo like we did in the previous example.
+
+Filter out the rest of the repo to isolate the needed gem folder:
+```
+cd achievements-gem
+git filter-repo --path Gems\Achievements
+```
+Now achievements-gem repo looks like:
+```
+root
+└───Gems
+    └───Achievements
+```
+We'll add the newly filtered source repo, achievements-gem, as a remote to our destination repo achievements-extras and we will refer to it as "achievements-gem":
+```
+cd ../achievements-extras
+git remote add achievement-gem ../achievements-gem
+```
+Update the remote:
+```
+git remote update
+```
+Create a new branch in the destination repo so we can PR this back in:
+```
+git switch -C move_achievements_gem_to_extras
+```
+Merge allowing unrelated histories:
+```
+git merge --allow-unrelated-histories achievements-gem/main
+```
+The branch should now have all the files with history in the correct location.
+
+Push the branch:
+```
+git push origin move_achievements_gem_to_extras
+```
+__Note: If there are LFS objects we need to upload them__

```

Comment:
The push will give an error if there are LFS objects; this is because the `git merge` will not result in copying over the large files, but instead their small hashfile representatives. As a workaround... before calling the 'git merge', copy the source repo's _.lfsconfig_ file into your destination folder. `git merge` will then copy over the actual files (ie: a gem preview.png will be a real PNG and not a text file containing a LFS hash number). From here, if the all the large file assets are less than 1GB then GitHub will provide enough space for LFS for free, otherwise, edit the `.lfsconfig > lfs.url` value to point to your new LFS server.